### DISCUSSION TITLE
What handling recommandation when receiving  ControllerException

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy#1000] [Q&A]
When sending a message to zigpy layer via request() or broadcast() what would be the proper way to handle the Exception  ControllerException

Should we simply stop the work and ask the end user to restart and unplug/plug its key ?

**Comment by Hedda** [comment]
Do you mean like having a [watchdog timer](https://en.wikipedia.org/wiki/Watchdog_timer) that will reset and/or restarts stuff similar to this PR by @puddly? -> https://github.com/zigpy/zigpy-deconz/pull/198

**Reply by pipiche38** [reply]
What I mean is that as part of zigpy library there are a set of potential Exceptions which can be through.
It would be important from a documentation and usage to get some understanding when the exceptions is through what is the context left and eventually what is expected to be done by the above layer
